apis/python/src/tiledb/vector_search/object_api/embeddings_ingestion.py

```diff
@@ -2,335 +2,476 @@
                     subprocess.check_call(
                         [sys.executable, "-m", "pip", "install", module]
                     )
 
         install_extra_modules()
+        import importlib
+
+        importlib.reload(tiledb)
 
-        import numpy as np
+        import logging
+        import math
+        import time
 
         import tiledb
+        from tiledb.cloud import dag
+        from tiledb.cloud.rest_api import models
+        from tiledb.cloud.utilities import get_logger
+        from tiledb.vector_search import ingest
+        from tiledb.vector_search.object_api import ObjectIndex
+        from tiledb.vector_search.object_readers import ObjectPartition
+
+        MAX_TASKS_PER_STAGE = 100
+        DEFAULT_IMG_NAME = "3.9-vectorsearch"
+        DEFAULT_WORKER_RESOURCES = {"cpu": "1", "memory": "4Gi"}
+
+        def setup(
+            config: Optional[Mapping[str, Any]] = None,
+            verbose: bool = False,
+        ) -> logging.Logger:
+            """
+            Set the default TileDB context, OS environment variables for AWS,
+            and return a logger instance.
+
+            :param config: config dictionary, defaults to None
+            :param verbose: verbose logging, defaults to False
+            :return: logger instance
+            """
+
+            try:
+                tiledb.default_ctx(config)
+            except tiledb.TileDBError:
+                # Ignore error if the default context was already set
+                pass
+
+            level = logging.DEBUG if verbose else logging.NOTSET
+            logger = get_logger(level)
+
+            logger.debug(
+                "tiledb.cloud=%s, tiledb=%s, libtiledb=%s",
+                tiledb.cloud.version.version,
+                tiledb.version(),
+                tiledb.libtiledb.version(),
+            )
 
-        def instantiate_object(code, class_name, **kwargs):
-            import importlib.util
-            import os
-            import random
-            import string
-            import sys
+            return logger
+
+        # --------------------------------------------------------------------
+        # UDFs
+        # --------------------------------------------------------------------
+        def compute_embeddings_udf(
+            object_reader_source_code: str,
+            object_reader_class_name: str,
+            object_reader_kwargs: Dict,
+            object_embedding_source_code: str,
+            object_embedding_class_name: str,
+            object_embedding_kwargs: Dict,
+            partition_dicts: List[Dict],
+            embeddings_uri: str,
+            metadata_array_uri: str = None,
+            index_timestamp: int = None,
+            verbose: bool = False,
+            trace_id: Optional[str] = None,
+            config: Optional[Mapping[str, Any]] = None,
+            extra_worker_modules: Optional[List[str]] = None,
+        ):
+            def install_extra_modules():
```

Comment:
(Minor)(Optional) Could we rename this to something else so it's easier to understand whether this one or the one above is being called? I.e. `install_extra_worker_modules` (and above could be `install_extra_driver_modules`?

Reply:
Applied